### DISCUSSION TITLE
Viewing commits from multiple branches?

### DISCUSSION THREAD

**Discussion by gugray** [jesseduffield/lazygit#3161] [Q&A]
I'm looking for a way to create the view that I'm used to from GitExtensions, which I find immensely useful. It's the view that shows all commits on all branches, "guitar hero" style. I seem to be unable to achieve this view in Lazygit. Is it possible?

This is the view in GitExtensions:
![image](https://github.com/jesseduffield/lazygit/assets/22029901/0b92a474-9647-48cc-87f2-82ea69a212d5)

In Lazygit, I always only see the commits from the currently checked out branch:
![image](https://github.com/jesseduffield/lazygit/assets/22029901/6d1fbb17-6850-4083-8889-a6c099c46956)

**Comment by mark2185** [comment]
You could always [supply your own `git log` command](https://github.com/jesseduffield/lazygit/blob/master/docs/Config.md#custom-git-log-command), would that help?

**Reply by gugray** [reply]
Thank you for responding! What command would I need to supply to get the multi-branch view of the timeline, and where can I do this in Lazygit?

**Reply by mark2185** [reply]
If I've googled it correctly, the required `git log` flag is `--all`, so I've just added that to the default value.

Add this to your `config.yml`:
```yml
git:
  branchLogCmd: "git log --graph --all --color=always --abbrev-commit --decorate --date=relative --pretty=medium {{branchName}} --"
```

**Reply by stefanhaller** [reply]
We already have a dedicated view to show the log of all branches; you can invoke it by pressing `1 a`. If you don't like the options that it uses, modify `git.allBranchesLogCmd` to customize it. (I prefer `--oneline` myself.)

**Reply by gugray** [reply]
Thank you for the answers, this is already becoming a lot clearer! (And sorry for the long hiatus.)

**Reply by mathias-lepoutre** [reply]
After reading this thread, let me share the setting I ended up with:
```
  allBranchesLogCmds:
    - "git log --graph --all --color=always --abbrev-commit --decorate --date=relative --pretty=format:'%C(yellow)%h %C(auto)%d %C(cyan)%an%C(reset): %s'"
    - git log --graph --all --color=always --abbrev-commit --decorate --date=relative  --pretty=medium
```
The second line is the default value, you can cycle through by pressing `a` again.
